amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -184,29 +198,55 @@ private void handleFlowError(Throwable t) {
         }
     }
 
-    private String throwableCategory(Throwable t) {
+    private enum ThrowableType {
+        ACUIRE_TIMEOUT("AcquireTimeout"), READ_TIMEOUT("ReadTimeout"), UNKNOWN("Unknown");
+
+        String value;
+
+        ThrowableType(final String value) {
+            this.value = value;
+        }
+    }
+
+    private class ThrowableCategory {
```

Comment:
You should make this class static as it doesn't need to close over `this`

Reply:
Addressed.